tests/testthat/test_MetaboLightsParam.R

```diff
@@ -1,26 +1,35 @@
 library(MsExperiment)
-test_that("Param is defined properly", {
-    expect_error(MetaboLightsParam(studyId = ")Qn"), "must start")
-    #study with only one assay: MTBLS10035
-    param <- MetaboLightsParam(studyId = "MTBLS10035")
+test_that("Object build properly", {
+    expect_error(MetaboLightsParam(mtblsId = ")Qn"), "must start")
+    ## Study with only one assay: MTBLS10035
+    param <- MetaboLightsParam(mtblsId = "MTBLS39")
     expect_is(param, "MetaboLightsParam")
     res <- readMsObject(MsExperiment(), param)
     expect_is(res, "MsExperiment")
     expect_is(res@sampleData, "DataFrame")
 
-    #also test a ID that does not work MTBLXXXX
-    param_test <- MetaboLightsParam(studyId = "MTBLSXXXX")
-    expect_error(readMsObject(MsExperiment(), param_test), "No assay files")
+    ## Test keepOntology and keepProtocol
+    res_filtered <- readMsObject(MsExperiment(), param,
+                                 keepOntology = FALSE,
+                                 keepProtocol = FALSE)
+    expect_lt(ncol(res_filtered@sampleData), ncol(res@sampleData))
 
-    # testing interactive sesh
-    mock_param <- MetaboLightsParam(studyId = "MTBLS8735")
+    ## Test simplify flag removes columns with NAs and duplicated columns
+    expect_true(all(colSums(is.na(res@sampleData)) != nrow(res@sampleData)))
+    expect_true(any(duplicated(as.list(res@sampleData))) == FALSE)
+})
+
+test_that("interactive session works", {
+    ## Testing interactive sesh
+    mock_param <- MetaboLightsParam(mtblsId = "MTBLS575")
     menu <- NULL
     with_mocked_bindings(
-        menu = function(choices, title = NULL) { 2 },
+        menu = function(choices, title = NULL) { 3 },
         {
             result <- readMsObject(MsExperiment(), mock_param)
         }
     )
-    expect_true(nrow(result@sampleData) == 10)
-    expect_true(ncol(result@sampleData) == 44)
+    expect_true(nrow(result@sampleData) == 6)
+    expect_true(ncol(result@sampleData) == 30)
 })
+
```

Comment:
please also add a unit test for `.clean_merged` - using some manually defined `data.frame` which allows you to test what you get is what you expect.

Reply:
Yes I forgot ! fixing the other things now and I will do that when we agree on the whole checking name thing  :)